rayon-core/src/join/mod.rs

```diff
@@ -32,7 +32,44 @@ mod test;
 /// while waiting for the thief to fully execute closure B. (This is the
 /// typical work-stealing strategy).
 ///
-/// ### Warning about blocking I/O
+/// # Examples
+///
+/// This example uses join to perform a quick-sort:
+///
+/// ```rust
+/// # use rayon_core as rayon;
+/// fn quick_sort<T:PartialOrd+Send>(v: &mut [T]) {
+///    if v.len() > 1 {
+///        let mid = partition(v);
+///        let (lo, hi) = v.split_at_mut(mid);
+///        rayon::join(|| quick_sort(lo),
+///                    || quick_sort(hi));
+///    }
+/// }
+///
+/// let mut v = vec![5, 1, 8, 22, 0, 44];
+/// quick_sort(&mut v);
+/// assert_eq!(v, vec![0, 1, 5, 8, 22, 44]);
+///
```

Comment:
This extra space will show up in the docs output.

Reply:
Question: do you think I should include the `partition` method? I expected there to be a link to play, but I don't think there actually is...?

Reply:
Decided to just uncomment it.